src/js/jsx/shared/Datalist.jsx

```diff
@@ -702,7 +706,11 @@ define(function (require, exports, module) {
                 );
             }
 
-            var dialog = searchableOptions && (
+            var filters = this.props.filters;
+            var dialog;
+            
+            if (filters && filters.indexOf(ADOBE_STOCK_FILTER) === -1) {
```

Comment:
It's definitely not acceptable for `Datalist` to have any knowledge of Adobe Stock. `Datalist` is a basic, shared component. This would be like Chrome adding Google-specific functionality to `<input type="text"/>`.